scripts/build-protoc-gen-dart

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+## Check dependencie(s)
+
+[[ $(command -v dart) ]] || { echo "'dart' not found!" ; dep_check="false" ;}
+[[ $(command -v wget) ]] || { echo "'wget' not found!" ; dep_check="false" ;}
+[[ $(command -v tar) ]] || { echo "'gzip' not found!" ; dep_check="false" ;}
+
+[[ ${dep_check} = "false" ]] && { echo "Some dependencie(s) isn't installed yet. Please install that dependencie(s)" ; exit 1 ;}
+
+## Variables
+protobuf_src="https://github.com/google/protobuf.dart/archive/refs/tags/protoc_plugin-19.2.0+1.tar.gz"
+setdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd)" # this line powered by stackoverflow
+kernelname="$(uname -s | tr '[:upper:]' '[:lower:]' || { echo 'kernel name can not definied' ; exit 1 ;})"
+
+# Defaults
+setfile="protoc_plugin/bin/protoc_plugin.dart"
+save_file="protoc-gen-dart_${kernelname}"
+
+# Check dir else create save dir
+if [[ $(basename ${setdir}) = "scripts" ]] ; then
+    if [[ $(basename $(dirname ${setdir})) = "starport" ]] ; then
+        [[ -d $(dirname "${setdir}/pkg/protoc-gen-dart/data") ]] || mkdir -p "$(dirname ${setdir})/starport/pkg/protoc-gen-dart/data"
+    else
+        echo "Attantion: you are running the script out of the startport project please run it this script in: https://github.com/tendermint/starport"
+        exit 1
+    fi
+else
+    echo "$setdir"
+    echo "Attantion: you are running the script out of the startport project please run it this script in: https://github.com/tendermint/starport"
+    exit 1
+fi
+
+## Check argument
+
+if ! [[ -z "${1}" ]] && [[ -d "${1}" ]]; then
+    save_dir="${1}"
+fi
+
+## Main
+# Check and Create Temp Directory
+[[ -d "/tmp/${0}" ]] && rm -rf "/tmp/${0}"
```

Comment:
I think we don't need to check before running `rm`, as rm command won't complain on non existent dirs.